File: Maga/1 курс/2 сем/ТОИБ/лекция 06/00vec.cpp
// Пример из лекций Алексея Кладова
// https://www.youtube.com/watch?v=Oy_VYovfWyo&list=PLlb7e2G7aSpTfhiECYNI2EZ1uAluUqE_e

#include <iostream>
#include <vector>

int main() {
    std::vector<int> xs = {1, 2, 3};
    auto &x = xs[0];
    printf("x = %d\n", x);

    xs.push_back(4);

    printf("x = %d\n", x);

    return 0;
}

File: Maga/1 курс/2 сем/ТОИБ/лекция 06/00vec.rs
// Пример из лекций Алексея Кладова
// https://www.youtube.com/watch?v=Oy_VYovfWyo&list=PLlb7e2G7aSpTfhiECYNI2EZ1uAluUqE_e

fn main() {
    let mut xs = vec![1, 2, 3];
    let x: &i32 = &xs[0];

    println!("x = {}", x);

    xs.push(4);

    println!("x = {}", x);
}
